### PR TITLE
Another MET resolution fix

### PR DESCRIPTION
**Pull request by jimbrooke**
This is resolves the issue with MHT reported by Zhenbin on 7 April.  The bug was introduced when I merged PR #230 by hand to create #231.

### CODE REVIEW

File: L1Trigger/L1TCalorimeter/src/firmware/Stage2Layer2EtSumAlgorithmFirmwareImp1.cc [hunk 1]
```diff
@@ -49,8 +49,8 @@ void l1t::Stage2Layer2EtSumAlgorithmFirmwareImp1::processEvent(const std::vector
         l1t::CaloTower tower = l1t::CaloTools::getTower(towers, ieta, iphi);
 
 	if (tower.hwPt()>metTowThresholdHw_ && CaloTools::mpEta(abs(tower.hwEta()))<=metEtaMax_) {
-	  ringEx += (int32_t) (tower.hwPt() * std::trunc ( 511. * cos ( 2 * M_PI * (72 - (iphi-1)) / 72.0 ) ));
-	  ringEy += (int32_t) (tower.hwPt() * std::trunc ( 511. * sin ( 2 * M_PI * (iphi-1) / 72.0 ) ));
+	  ringEx += (int32_t) (tower.hwPt() * std::trunc ( 1023. * cos ( 2 * M_PI * (72 - (iphi-1)) / 72.0 ) ));
```
Comment (fwyzard) [inline L1Trigger/L1TCalorimeter/src/firmware/Stage2Layer2EtSumAlgorithmFirmwareImp1.cc:52]: isn't 
`cos ( 2 * M_PI * (72 - (iphi-1)) / 72.0 )`
the same as
`cos ( 2 * M_PI * (iphi-1) / 72.0 )`
?

**Comment by fwyzard** [comment]
hi Jim,
does this affect only ETM, or also ETT/HTM ? 

Looking at the slides 5 and 6 of https://indico.cern.ch/event/515695/contribution/5/attachments/1254459/1851242/Zhenbin_TSGv4p1.pdf , are the "floating" results correct ?

File: L1Trigger/L1TCalorimeter/src/firmware/Stage2Layer2EtSumAlgorithmFirmwareImp1.cc (hunk 1 above)
Comment (jimbrooke) [inline L1Trigger/L1TCalorimeter/src/firmware/Stage2Layer2EtSumAlgorithmFirmwareImp1.cc:52]: I believe so.  I have a long-standing intention to follow up with the original author of this (who also wrote the firmware) to find out why it was coded like this in the first place.

**Comment by jimbrooke** [comment]
There are two changes here.  The first corrects a factor 2 error in HTM.

The second is a smaller change to ETM that corrects the number of bits used for trig coefficients, from 9 to 10.  This will bring the integer calculation slightly closer to floating point, but it is a marginal change.

For HTM, the floating point result in the slides is correct.  I can't comment on ETT yet.

**Comment by rekovic** [comment]
This PR does seems to have only a minor affect, almost unnoticeable, on ETM but 
it has a very noticeable effect on HTM making the distribution harder.
(HTT and ETT are not affected.)

http://rekovic.web.cern.ch/rekovic/cms/L1T/Stage2/l1t-integration/compare_test_l1t_integration_v27.0_vs_test_l1t_integration_v26.1/

**Comment by jimbrooke** [comment]
@rekovic great, that is what I expect :)

**Comment by rekovic** [comment]
@jimbrooke This will make HTM rates go up. 
l1t-integration-v26 ntuples should be available. We need to see rates done with those, like the right-hand side plot of slide 6 from his presentation yesterday [1], to understand where this PR puts us with respect to red-and-black line difference in his new plot.  This assumes we understand the red line and take it as the correct one.  
Mind you that slide 6 of the Zhenbin's presentation [1] shows HTM emulate rates already higher than the red line. 

[1] https://indico.cern.ch/event/515695/contribution/5/attachments/1254459/1851242/Zhenbin_TSGv4p1.pdf

**Comment by fwyzard** [comment]
I assumed that the _black_ lines are correct, and that disagreement with the red lines points to bug(s) in the emulator.
This pr should likely fix HTM and ETM to agree with the black lines.

**Comment by fwyzard** [comment]
+1 to have it in the next l1t-tsg tag.

**Comment by rekovic** [comment]
I confused the red and black line.  The red line is the emulator, OK and black line Zhenbins on-fly calculation. 
This PR will have emulator rates go up in the direction of black line. 

Agree with +1.

**Comment by rekovic** [comment]
Tagges as l1t-integration-v27.0

**Pull request merged**